docs/rule/no-builtin-form-components.md

```diff
@@ -51,6 +53,8 @@ You may consider composing the [set helper](https://github.com/pzuraq/ember-set-
 />
 ```
 
+Depending on your requirements, consider using form management patterns like the "light" [Form component from ember-primitives](https://ember-primitives.pages.dev/7-forms/1-intro) or the "controlled" [ember-headless-form](https://ember-headless-form.pages.dev/).
```

Comment:
this isn't quite what I meant about adding. :sweat_smile: 

[This is docs, for example](https://limber.glimdown.com/edit?c=MQAgYg9gTgtgzgKAcFICqcCWA7A5iAfQBUALAUxAAUAbAQwBcAzaGApANVu02rpAClaAN1oBlAMZRMAB3ogStOCDJCyUAJ70SOfAHcK2MmQAmIehHldj1Cs1ghjDWgBoHZALRx12cSExzGKAgYEAgAVygQOHCocQpHeloQKxBxCF4ycTlaXhAwuDU-bGkwuTDpCGwosIAjGEw4LEqAOiQAeS1C2hq4eihaLMxKuFc4MPESZKVOkAAiCa5cMgL6WdTKuNl17EScZJBqCDkIRjNtJUYwn3oh7Bz-dWTsUwVVZONjf1ucvJvqL%2BW1QmUxAQV4NQGAGtRndpHASEcRiA7JCdK4yPRxK4aqUzgw-EokuZpJhfHYnkdyJFjGQYBBWggAJJVGZ2EI1MiHXSufSpKBkBgUJKcbi8WiAIkIQABtAASRAAsgAZZEsAC6UoATKrXClKtRHrQPnMAKIwDlQNZpGkqyIKZ7-PBnWwsap1BpNKopRiYTmmHAlehwZogABCjxpjFoYWo9FcbNd9UatyUuh41FSUYK%2B2ktCWoM5EENowseySWDwNncOaW7lo0mkPIMRlM5hA0n5qh2eLkEajMZAHNeQygDIwOidIDuNzeAEFKIzXFLILAACJOdUARm1IF54i4IFwDCpE-yhX9pRT5BZ5DyBUi7eWBWmN7GbrkOPo5mwDIABn-cAAVko-yqAgmAwBUUByAA3qEVQAL7IkEIQAOQAAK0uaAD0dKfN6agoQA3GBEHQDBqScumiGBMEIAoZhajuPy0QRHEcBEUgaTYL0USJPQFAALwUbwAAUACUxEIFxPF2tYZDMgGIBCSJKhkDsYlKQAfCA0EICABwYjaMBrokSmTmQujgCwJm0Cpnb0M04gRPyOxELQUBLPQEl6QZPZOGZbQ1ABmQOTRMDGjsUjLCJbI2c0al9D6cDid5Pm9IKjnOQlZn8KIbQAHLNL0Uh4JgjDqCJCQuJO0bUK4GrefBknSXIsk2KItT1HIynKPZGkCdpun6apOzNA%2B9krmQkbRvQ4nEfpbXycUpR2QljWSQAPPxEF0Pxmk%2BRtbI%2Bfp0HQZUdHnvQKGWPaS0BvB8HHTpZ1VChr5dddi0dW%2BD1PeIdCNAJsy4FIpiHtI7garMT29OoNhAzAtAAB7uKmxhaAAXCAADMAAMuPSEj0P6ft%2Bn6RtdActQ2lgJgUA8fltAwGQT3k5dk5M2QAkod69P0IzzMoaTZMbVhlOcvtT0U90EvgMI0D%2BBQJks2TIvs-Q6jSFzKEJGQ113Mz3ORkICv8crQtS2LMvU0gqsbR%2BX5mJr2vvf4QvfV1osO5Uwui2ykvkzUUCafbwe21Zq5OBjB0Pppp3pfxmVQC59APaLsfhxtsM2ML%2Bns7BNTQDSUBYxuhNROkmDGIRICPeTWHZ2Q%2B2i9t0i7U3CB-j%2BSAbSQYfIKgsw0AwbJrAASlNahqWxSDuNKP6HSwmk-nBIDyiu%2BXqlqCBz1KP7LsZTgr%2Bd6%2Bb1KW5ICgoCmuaIAT4wU8%2BMss-SjMP6VCvuFlT6UDqtjqpIHPqqLGJBPxwgxlhLCNJVCHC1iOOkAAvNMtBmjQFwFhNS7g0CiCgRAcQcAsIAHUyA1CwnORkWED42QQJqYB8gwFwAgVA1SsC1DNEQcg1BHkMHYCwTg4weCCHENIXKJUWFjQ2GZjsLCR0pT-xAQwphuAwhV2WPFM0aggKOWCFhfkNhFBkCwmkUihgdgEOMRUUx9BPB8Q8FYWsgxhhYWAOYcQBBQEwGoO4L%2B%2BF6a1meA4m4wwEBAA&format=glimdown)

I think we should show an "uncontrolled" example before the controlled example. I think it's important that folks see how to _not_ write JavaScript before they reach for javascript.

Reply:
Ahh yes. That makes a lot more sense 😅 

FWIW I appreciate the resources you're creating and the modern improvements.

But we should have empathy for the frustrated engineer who is on the receiving end of lint errors, and try to give them a very clear explanation of the problem and how to solve it.

That's why I'm not using the new component authoring format in the examples, too.

Reply:
> FWIW I appreciate the resources you're creating and the modern improvements.

<3 


> other new ideas.

controlling every aspect of a form is actually the new idea -- the uncontrolled way is the very old way to do forms, it builds on the foundation of the platform and allows the developer to do less work.

Additionally, React has been trying to teach uncontrolled vs controlled forms for 5+ years! :muscle: 

> That's why I'm not using cell or the new component authoring format in the examples.

that's totally fine! the _details_ are a means to the concept, and not actually important to the concept -- I think that's probably one thing people get too hung up on when learning new things.  It's important to be able to adapt a concept in to your existing code, as not all documentation can match what your existing code is doing.

> Anyway I had another run at it. Let me know your thoughts

will do, thanks a ton for working on this and being willing to have the back and forth!!

Reply:
> a means to the concept, and not actually important to the concept
> can match what your existing code is doing.

which you've don't exactly! yay!